### PR TITLE
Added VolumeMount support

### PR DESCRIPTION
**Pull request by solarkennedy**
This adds VolumeMount support to the main container and extraContainers,
so that they can actually use the Volumes defined in the previous PR.

### CODE REVIEW

**Comment by solarkennedy** [comment]
This one is ready for review too, but I'm a bit concerned.

For this one, it seems to complained that all existing jobs on my dev stack 

```2021-09-25 01:03:30,520 ERROR [cass_titus_dev-nio-worker-3] service.JobReconciliationFrameworkFactory:? - Bad job record found: jobId=83482f6f-338f-40b8-9383-4def4709699f, violations={jobDescriptor.container='null' in fields: volumeMounts}```

Which makes me think I *must* relax the nonNull constraint?

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/BasicContainer.java [hunk 1]
```diff
@@ -54,18 +54,22 @@
     )
     private final Map<String, String> env;
 
+    private final List<VolumeMount> volumeMounts;
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/BasicContainer.java:57]: This class misses validation annotations.

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/BasicContainer.java [hunk 2]
```diff
@@ -77,20 +81,30 @@ public Image getImage() {
...
-        if (command == null) { return Collections.emptyList();}
+        if (command == null) {
+            return Collections.emptyList();
+        }
         return command;
     }
 
     public Map<String, String> getEnv() {
-        if (env == null) { return Collections.emptyMap();}
+        if (env == null) {
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/BasicContainer.java:98]: Validation annotations should ensure that all values are non-null.

**Review by tbak — CHANGES_REQUESTED** [review]

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/BasicContainer.java (hunk 2 above)
Comment (solarkennedy) [reply titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/BasicContainer.java:98]: Gotcha, I've removed these from command/env/entrypoint.

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/BasicContainer.java (hunk 1 above)
Comment (solarkennedy) [reply titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/BasicContainer.java:57]: Added `CollectionInvariants`

**Comment by solarkennedy** [comment]
I can confirm now that this branch correctly creates volumeMounts on the pod object, and is roll-backsafe (jobs with volumemounts still can be read by old code, of course you can't see the volumeMounts in the json, but the code still works).

**Review by tbak — APPROVED** [review]

**Pull request merged**